yb-voyager/src/cp/yugabyted/yugabyted.go

```diff
@@ -0,0 +1,593 @@
+
+		if updateExportedRowCountEvent.Status == "COMPLETED" {
+			updateExportedRowCountEvents = append(updateExportedRowCountEvents, &updateExportedRowCountEvent.BaseUpdateRowCountEvent)
+		} else if !check {
+			cp.lastRowCountUpdate[updateExportedRowCountEvent.TableName] = time.Now()
+
+			updateExportedRowCountEvents = append(updateExportedRowCountEvents, &updateExportedRowCountEvent.BaseUpdateRowCountEvent)
+		} else if !lastRowCountUpdateTime.Add(time.Second * 5).After(time.Now()) {
+			cp.lastRowCountUpdate[updateExportedRowCountEvent.TableName] = time.Now()
+
+			updateExportedRowCountEvents = append(updateExportedRowCountEvents, &updateExportedRowCountEvent.BaseUpdateRowCountEvent)
+		}
+
+	}
+	cp.createAndSendUpdateRowCountEvent(updateExportedRowCountEvents)
+}
+
+func (cp *YugabyteD) ImportSchemaStarted(importSchemaEvent *controlPlane.ImportSchemaStartedEvent) {
+	cp.createAndSendEvent(&importSchemaEvent.BaseEvent, "IN PROGRESS", "")
+}
+
+func (cp *YugabyteD) ImportSchemaCompleted(importSchemaEvent *controlPlane.ImportSchemaCompletedEvent) {
+	cp.createAndSendEvent(&importSchemaEvent.BaseEvent, "COMPLETED", "")
+}
+
+func (cp *YugabyteD) SnapshotImportStarted(snapshotImportEvent *controlPlane.SnapshotImportStartedEvent) {
+	cp.createAndSendEvent(&snapshotImportEvent.BaseEvent, "IN PROGRESS", "")
+}
+
+func (cp *YugabyteD) SnapshotImportCompleted(snapshotImportEvent *controlPlane.SnapshotImportCompletedEvent) {
+	cp.createAndSendEvent(&snapshotImportEvent.BaseEvent, "COMPLETED", "")
+}
+
+func (cp *YugabyteD) UpdateImportedRowCount(
+	snapshotImportTableMetrics []*controlPlane.UpdateImportedRowCountEvent) {
+
+	cp.Mutex.Lock()
+	defer cp.Mutex.Unlock()
+
+	var updateImportedRowCountEvents []*controlPlane.BaseUpdateRowCountEvent
+	for _, updateImportedRowCountEvent := range snapshotImportTableMetrics {
+
+		lastRowCountUpdateTime, check := cp.lastRowCountUpdate[updateImportedRowCountEvent.TableName]
+
+		if updateImportedRowCountEvent.Status == "COMPLETED" {
+			updateImportedRowCountEvents = append(updateImportedRowCountEvents, &updateImportedRowCountEvent.BaseUpdateRowCountEvent)
+		} else if !check {
+			cp.lastRowCountUpdate[updateImportedRowCountEvent.TableName] = time.Now()
+			updateImportedRowCountEvents = append(updateImportedRowCountEvents, &updateImportedRowCountEvent.BaseUpdateRowCountEvent)
+		} else if !lastRowCountUpdateTime.Add(time.Second * 5).After(time.Now()) {
+			cp.lastRowCountUpdate[updateImportedRowCountEvent.TableName] = time.Now()
+			updateImportedRowCountEvents = append(updateImportedRowCountEvents, &updateImportedRowCountEvent.BaseUpdateRowCountEvent)
+		}
+
+	}
+	cp.createAndSendUpdateRowCountEvent(updateImportedRowCountEvents)
+}
+
+func (cp *YugabyteD) MigrationEnded(migrationEndedEvent *controlPlane.MigrationEndedEvent) {
+}
+
+func (cp *YugabyteD) panicHandler() {
+	if r := recover(); r != nil {
+		// Handle the panic for eventPublishers
+		log.Errorf(fmt.Sprintf("Panic occurred:%v\n"+
```

Comment:
Indicate in the log message that further event publishing will be stopped by the go-routine.

Reply:
Done.